src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java

```diff
@@ -64,38 +83,78 @@ public User findByEmail(String email) {
                 .orElseThrow(() -> new UserNotFoundException("email"));
     }
 
+    public AuthenticationResponse updateAuthenticateUser(Long id, UserDTO userDTO) throws ChangeSetPersister.NotFoundException {
+        User existingUser = userRepository.findByIdAndDeletedFalse(id).orElseThrow(ChangeSetPersister.NotFoundException::new);
+        userValidations(userDTO);
+        User updatedUser = modelMapper.map(userDTO, User.class);
+        updatedUser.setId(existingUser.getId());
+        userRepository.save(updatedUser);
+
+
+        String jwtToken = jwtService.generateToken(updatedUser);
+        String refreshToken = jwtService.generateRefreshToken(updatedUser);
+
+        tokenService.revokeAllUserTokens(updatedUser);
+        tokenService.saveToken(updatedUser, jwtToken, TokenType.ACCESS);
+        tokenService.saveToken(updatedUser, refreshToken, TokenType.REFRESH);
+
+        return AuthenticationResponse
+                .builder()
+                .accessToken(jwtToken)
+                .refreshToken(refreshToken)
+                .user(modelMapper.map(updatedUser, PublicUserDTO.class))
+                .build();
+
+
+    }
+
     public UserDTO updateUser(Long id, UserDTO userDTO) throws ChangeSetPersister.NotFoundException {
         User existingUser = userRepository.findByIdAndDeletedFalse(id).orElseThrow(ChangeSetPersister.NotFoundException::new);
-        if (StringUtils.isBlank(userDTO.getUsername())) {
-            throw new ValidationException("Username is required");
-        }
-        if (StringUtils.isBlank(userDTO.getAddress())) {
-            throw new ValidationException("Address is required");
-        }
-        if (StringUtils.isBlank(userDTO.getFirstname())) {
-            throw new ValidationException("First Name is required");
+        userValidations(userDTO);
+        User updatedUser = modelMapper.map(userDTO, User.class);
+        updatedUser.setId(existingUser.getId());
+        userRepository.save(updatedUser);
+        return modelMapper.map(updatedUser, UserDTO.class);
+
+    }
+
+    public boolean ifPasswordMatch(String password) {
+        UserDTO authenticatedUserDTO = findAuthenticatedUser();
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        boolean ifMatch = passwordEncoder.matches(password, authenticatedUserDTO.getPassword());
+        if (ifMatch) {
+            return true;
+        } else {
+            return false;
         }
```

Comment:
you can shorten the method significantly like this -**_public boolean ifPasswordMatch(String password) {
    UserDTO authenticatedUserDTO = findAuthenticatedUser();
    BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
    return passwordEncoder.matches(password, authenticatedUserDTO.getPassword());_**
}